plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstance.java

```diff
@@ -306,6 +308,60 @@ public DockerNode getNode() {
         return node;
     }
 
+    /**
+     * Reads file content by specified file path.
+     *
+     * TODO:
+     * add file size checking,
+     * note that size checking and file content reading
+     * should be done in an atomic way,
+     * which means that two separate instance processes is not the case.
+     *
+     * @param filePath
+     *         path to file on machine instance
+     * @param startFrom
+     *         line number to start reading from
+     * @param limit
+     *         limitation on line
+     * @return all file content if {@code startFrom} is 0 or less,
+     * if {@code limit} grater than 0 and {@code startFrom} less than {@code limit} and
+     * grater than 0 content from {@code startFrom} to {@code startFrom +limit} will be returned,
+     * if {@code limit} is equal to 0 or less then {@code startFrom} will be returned only line with number {@code startFrom},
+     * if file contains less lines than {@code startFrom} empty content will be returned
+     * @throws MachineException
+     *         if any error occurs with file reading
+     */
+    @Override
+    public String readFileContent(String filePath, int startFrom, int limit) throws MachineException {
+        if (limit < 0) {
+            throw new MachineException("Impossible to read " + limit + " lines");
+        }
+
+        FormattedLineConsumer fileContent = new FormattedLineConsumer();
+        InstanceProcess process;
+        String command;
+        try {
+            if (startFrom > 0) {
+                // command sed getting file content from startFrom to startFrom + limit.
+                command = String.format("sed -n \'%1$2s, %2$2sp\' %3$2s", startFrom, startFrom + limit, filePath);
+            } else {
+                //getting all file content
+                command = "cat " + filePath;
+            }
+            process = createProcess(command);
```

Comment:
I would prefer usage of docker exec instead of DockerProcess. In that case it won't be presented in list of processes, and you won't need workaround in 100-102 lines of DockerProcess.